File: Data Structures & Algorithms/sort-colors/submission-0.java
class Solution {
    public void sortColors(int[] nums) {
    
    int st = -1;
    int i = 0;
    int end = nums.length ;

    while(i < end){

        if(nums[i] == 0){
            st++;
            int temp= nums[st];
            nums[st] = nums[i];
            nums[i]= temp;
            i++;
        }

        else if(nums[i] == 1){
            i++;
        }
        else{
            end --;
            int temp = nums[end];
            nums[end] = nums[i];
            nums[i] = temp;
        }
    }
        
    }
}

File: Data Structures & Algorithms/contains-duplicate-ii/submission-0.cs
public class Solution {
    public bool ContainsNearbyDuplicate(int[] nums, int k) {
        var map = new Dictionary<int,int>();

        for(int i=0; i<nums.Length;i++){
            int key = nums[i];

            if(map.ContainsKey(key)){

                if (Math.Abs(map[key]-i) <= k ) return  true; 
            }
            map[key] = i;
        }
        return false;
    }
}